### PR TITLE
GUI: Get rid of some deprecated calls.

### PR DESCRIPTION
**Pull request by sinetek**
* QSslSocket -> QSslConfiguration

* Use std::lower_bound() and std::upper_bound() instead of Qt ones.

* Use std sorts instead of Qt ones.

* TextWidth will use the right Qt function depending on version.

* primaryScreen()->geometry() is now what we use.

* desktop() -> primaryScreen().

### CODE REVIEW

**Comment by sinetek** [comment]
Ready for review.

File: src/qt/paymentserver.cpp [hunk 1]
```diff
@@ -648,7 +648,7 @@ void PaymentServer::fetchPaymentACK(WalletModel* walletModel, const SendCoinsRec
         qWarning() << "PaymentServer::fetchPaymentACK: Error getting refund key, refund_to not set";
     }
 
-    int length = payment.ByteSize();
+    int length = payment.ByteSizeLong();
```
Comment (CaveSpectre11) [inline src/qt/paymentserver.cpp]: This change isn't building... 

> qt/paymentserver.cpp:651:26: error: ‘class payments::Payment’ has no member named ‘ByteSizeLong’; did you mean ‘ByteSize’?
     int length = payment.ByteSizeLong();
                          ^~~~~~~~~~~~
                          ByteSize

Comment (sinetek) [reply src/qt/paymentserver.cpp]: what version of protobuf do you have?

Comment (sinetek) [reply src/qt/paymentserver.cpp]: fixed it

File: src/libzerocoin/bignum.h [hunk 2]
```diff
@@ -10,6 +10,7 @@
 #include "veil-config.h"
 #endif
 
+#include <limits.h>
```
Comment (CaveSpectre11) [inline src/libzerocoin/bignum.h]: This looks to be successful (limits must have been getting pulled in somewhere down the path of compat.h).

Let's put this into a separate PR so we can fast-track it through before anything else.

**Comment by mimirmim** [comment]
Travis looks to be failing.

**Comment by CaveSpectre11** [comment]
> Travis looks to be failing.

It's failing for the timeout (test taking too long).  For some reason my request to requeue the tests aren't taking.  I'm not sure why.

**Comment by codeofalltrades** [comment]
utACK
All travis build have passed

**Comment by CaveSpectre11** [comment]
utACK https://github.com/Veil-Project/veil/pull/730/commits/fec6e0aea2664313a11ef157511a91be3b71a6c9
